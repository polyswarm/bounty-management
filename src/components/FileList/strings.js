const strings = {
  title: 'Files',
  clearAll: '- Clear',
};
export default strings;
