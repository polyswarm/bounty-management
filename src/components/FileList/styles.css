.File-List-Header {
  border-bottom: 2px solid #350066;
  color: #7500ff;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  font-size: 20px;
  line-height: 40px;
  align-items: center;
}

.File-List {
  background-color: #fafafa;
  border: 1px solid #8500ff;
  display: inline-block;
  overflow: auto;
  position: relative;
}

.File-List-Title {
  display: inline-block;
}

.List {
  color: #000;
}

.File-List-Header > button {
  background-color: #ff1a00;
  border-radius: 2px;
  color: white;
  float: right;
  margin-right: 15px;
  max-width: 120px;
  padding: 0;
}

.File-List-Header > button:hover {
  background-color: #ff8c75;
}
