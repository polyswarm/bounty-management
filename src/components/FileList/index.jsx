// Vendor imports
import React from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import FileProgress from '../FileProgress';
// Component Imports
import strings from './strings';
import './styles.css';

class FileList  extends React.Component {
  constructor(props) {
    super(props);
    this.onRemoveClickHandler = this.onRemoveClickHandler.bind(this);
  }

  render () {
    const { props: { files } } = this;
    return (
      <div className='File-List'>
        <header>
          {strings.title}{' '}{files.length}
        </header>
        <div className='List'>
          {files && files.length > 0 && (
            files.map((f, index) => {
              const name = f.name;
              return(
                <FileProgress
                  key={name}
                  file={f}
                  remove={() => {this.onRemoveClickHandler(index);}} />
              );
            })
          )}
        </div>
      </div>
    );
  }

  onRemoveClickHandler(index) {
    const { props: { removeFile } } = this;
    if (removeFile) {
      removeFile(index);
    }
  }
}

FileList.proptypes = {
  files: PropTypes.array.isRequired,
  removeFile: PropTypes.func.isRequired,
};

export default FileList;
