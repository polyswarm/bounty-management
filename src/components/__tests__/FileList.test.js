import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson, mountToJson} from 'enzyme-to-json';
import FileList from '../FileList';

it('throws when created without files', () => {
  expect(() => { render(<FileList />);  }).toThrow();
});

it('renders without crashing', () => {
  const files = [];
  const wrapper = render(<FileList files={files} />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('contains 3 FileProgress when 3 added', () => {
  const files = [
    {name: 'asdf'},
    {name: 'demo'},
    {name: 'test'},
  ];
  const wrapper = render(<FileList files={files} />);
  expect(wrapper.find('.ListItem')).toHaveLength(3);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('calls remove when FileProgress remove button clicked',() => {
  const remove = jest.fn();
  const files = [
    {name: 'asdf'},
  ];
  const wrapper = mount(
    <FileList
      removeFile={remove}
      files={files} />
  );
  wrapper.find('.ListItem').simulate('mouseEnter');
  wrapper.find('button').simulate('click');
  expect(remove).toHaveBeenCalled();
  expect(mountToJson(wrapper)).toMatchSnapshot();
});
