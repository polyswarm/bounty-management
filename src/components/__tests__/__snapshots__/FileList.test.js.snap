// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls remove when FileProgress remove button clicked 1`] = `
<FileList
  files={
    Array [
      Object {
        "name": "asdf",
      },
    ]
  }
  removeFile={[Function]}
>
  <div
    className="File-List"
  >
    <header
      className="File-List-Header"
    >
      Bounty Files
       
      1
    </header>
    <div
      className="List"
    >
      <ListItem
        item="asdf"
        key="asdf"
        remove={[Function]}
      >
        <div
          className="ListItem"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <p
            className="ListItem-Name"
          >
            asdf
          </p>
          <span
            className="ListItem-Remove"
          >
            <RemoveButton
              onClick={[Function]}
            >
              <button
                className="Remove-Button"
                onClick={[Function]}
              >
                <img
                  alt="remove"
                  className="redx"
                  src="/img/red-x.svg"
                />
              </button>
            </RemoveButton>
          </span>
        </div>
      </ListItem>
    </div>
  </div>
</FileList>
`;

exports[`contains 3 FileProgress when 3 added 1`] = `
<div
  class="File-List"
>
  <header
    class="File-List-Header"
  >
    Bounty Files 3
  </header>
  <div
    class="List"
  >
    <div
      class="ListItem"
    >
      <p
        class="ListItem-Name"
      >
        asdf
      </p>
    </div>
    <div
      class="ListItem"
    >
      <p
        class="ListItem-Name"
      >
        demo
      </p>
    </div>
    <div
      class="ListItem"
    >
      <p
        class="ListItem-Name"
      >
        test
      </p>
    </div>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  class="File-List"
>
  <header
    class="File-List-Header"
  >
    Bounty Files 0
  </header>
  <div
    class="List"
  >
    
  </div>
</div>
`;
